backend/error_source.go

```diff
@@ -0,0 +1,31 @@
+package backend
+
+import "net/http"
+
+// ErrorSource type defines the source of the error
+type ErrorSource string
+
+const (
+	ErrorSourcePlugin     ErrorSource = "plugin"
+	ErrorSourceDownstream ErrorSource = "downstream"
+)
+
+// GetErrorSource returns a error source based on http status code
+func GetErrorSource(statusCode int) ErrorSource {
```

Comment:
Nit. GetErrorSource feels generic and not specifically mentioning HTTP status so suggest the following

```suggestion
// ErrorSourceFromStatus returns an ErrorSource based on provided HTTP status code.
func ErrorSourceFromHTTPStatus(statusCode int) ErrorSource {
```